### PR TITLE
[PKG-2859] recordlinkage 0.16 ❄️ 

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/J535D165/recordlinkage/releases
License: https://github.com/J535D165/recordlinkage/blob/v0.16/LICENSE
Requirements: https://github.com/J535D165/recordlinkage/blob/v0.16/pyproject.toml

Actions:
1. Add `--no-deps --no-build-isolation` to script
2. Add missing `setuptools`, `setuptools_scm`, and `wheel` to `host`
3. Update `run` dependencies
4. Add `run_constrained`
5. Add `pip check`
6. Add `dev_url`
7. Add `description`

Notes:
- To be removed before merging [abs.yaml](https://github.com/AnacondaRecipes/recordlinkage-feedstock/pull/2/files#diff-15d5a29fc73da2888a21d894110d912b33b522513219d638e037e528179bb6ec)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,47 +1,60 @@
...
+    - wheel
   run:
-    - python >=3.5
-    - numpy
+    - python
+    - jellyfish >=1
+    - joblib
+    - numpy >=1.13
+    # pandas already has numexpr and bottleneck as dependencies
     - pandas >=0.18.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - pandas >=1,<3
```
https://github.com/J535D165/recordlinkage/blob/v0.16/pyproject.toml#L20C12-L20C18

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,47 +1,60 @@
...
-    - pandas >=0.18.0
-    - scipy
-    - scikit-learn
-    - jellyfish
-    - numexpr
+    - python
+    - jellyfish >=1
+    - joblib
+    - numpy >=1.13
+    # pandas already has numexpr and bottleneck as dependencies
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: What is this comment for? Is it because both are specified in the "all" extra?

**Review by JeanChristopheMorinPerso — APPROVED** [review]
LGTM, but I would run the test suite I think. From the look of it, it seems relatively lightweight.

**Comment by skupr-anaconda** [comment]
> LGTM, but I would run the test suite I think. From the look of it, it seems relatively lightweight.

The `tests` folder isn't available inside the Source Distribution on PyPi https://pypi.org/project/recordlinkage/#files. I would not change the source URL to the GitHub one

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I modified the comment and moved it to the `run_constrained` section

**Pull request merged**